### PR TITLE
Add test documenting #17230

### PR DESCRIPTION
**Pull request by tamird**
Add a test demonstrating #17230.

### CODE REVIEW

**Comment by tamird** [comment]
@hauntsaninja

**Comment by tamird** [comment]
@hauntsaninja would you mind taking a look?

**Comment by JelleZijlstra** [comment]
I'm not Shantanu but I don't think we should add a test that's testing incorrect behavior. Maybe make it an xfail test with the desired behavior.

**Comment by tamird** [comment]
How do I make it xfail?

**Comment by JelleZijlstra** [comment]
Grep for `xfail` in the mypy sourcecode and you'll see.

File: test-data/unit/check-python310.test [hunk 1]
```diff
@@ -1220,6 +1220,21 @@ def main() -> None:
...
             reveal_type(a)  # N: Revealed type is "builtins.int"
 
+[case testMatchCapturePatternFromAsyncFunctionReturningUnion-xfail]
+async def func1(arg: bool) -> str | int: ...
+async def func2(arg: bool) -> bytes | int: ...
+
+async def main() -> None:
+    match await func1(True):
+        case str(a):
+            match await func2(True):
```
Comment (JelleZijlstra) [inline test-data/unit/check-python310.test:1230]: Does this nesting add anything to the test? Better to keep tests simple unless the additional complexity tests something interesting.

Comment (tamird) [reply test-data/unit/check-python310.test:1230]: I was following the lead of the preceding test case. See above ^

**Comment by tamird** [comment]
Done, thanks.

**Comment by tamird** [comment]
@JelleZijlstra anything further?

**Comment by tamird** [comment]
Ping. Can we merge this please?

**Comment by JelleZijlstra** [comment]
It's on my list of pending reviews, so I will get to it.

However, the repeated pings feel very strange to me. Why is it important to you that we merge a change that just adds a failing test?

**Pull request merged**